### ISSUE TITLE
Need to update kroxylicious-samples instructions

### ISSUE THREAD

**Issue by SamBarker** [kroxylicious/kroxylicious#536]
**kroxylicious-sample**
- For the build commands you shouldn't need to run `mvn clean install`, `mvn package` or `mvn verify` should suffice

- The "dist" profile doesn't seem to exist, so I wasn't sure how to build executable JAR files
- The java command states I need to point to a classpath, but it's unclear which directory this would be
- I wasn't able to get the sample working due to the above two problems

_Originally posted by @katheris in https://github.com/kroxylicious/kroxylicious/discussions/532_

**Comment by SamBarker** [comment]
> The "dist" profile doesn't seem to exist, so I wasn't sure how to build executable JAR files
Dist is definitely still with us in the [kroxylicious pom](https://github.com/kroxylicious/kroxylicious/blob/main/kroxylicious/pom.xml#L293) however its not part of the sample it self. 

We should look at making the language clearer or having the sample-depend on an executable artefact. 

Without `dist` running classpaths and what to execute gets very confused.

**Comment by SamBarker** [comment]
We should also look at using the maven exec plugin (or something similar) for avoiding the the class path wrangling.  As suggested on [Slack](https://kroxylicious.slack.com/archives/C04V1K6EAKZ/p1691499787267119?thread_ts=1691491114.162329&cid=C04V1K6EAKZ)

**Comment by katheris** [comment]
> Dist is definitely still with us in the [kroxylicious pom](https://github.com/kroxylicious/kroxylicious/blob/main/kroxylicious/pom.xml#L293) however its not part of the sample it self.

Looking back at my commands I had already built the whole project, so I tried to run `mvn package -pl kroxylicious-sample -Pdist -Dquick`, so that explains why I couldn't see the profile

**Comment by gracegrimwood** [comment]
> So when consuming via the proxy a string that isn't structured as JSON anything with "bar" or "baz", and anything with a space gets lost. Sending a JSON structured string with "baz" and sending something that sort of looks like JSON at the beginning but isn't also gets lost.

@katheris That's really interesting (and a bit concerning!). I didn't observe anything like that while writing the integration tests for the sample, so I wonder if this is a difference between how the tests work and how running it works in reality. Could you please try commenting out the filters in [kroxylicious-sample/sample-proxy-config.yml](https://github.com/kroxylicious/kroxylicious/blob/main/kroxylicious-sample/sample-proxy-config.yml)? It would be interesting to see if the same behaviours are observed when removing one filter, then the other, then both.

My current hypothesis is that given how the integration tests work, it seems unlikely that the problem is with the filters failing to forward the request/response. If that were the case the tests would surely fail, so I wonder if something else is happening beyond the filters/within Kroxylicious to cause this behaviour. Especially given the sample filters don't care about JSON at all and treat everything as strings, so there's definitely no obvious reason for those to go missing. Additionally, any failure of a filter should produce an error, and this appears to be happening silently.

This is a very uneducated guess, but this could be something to do with how we're handling values returned from filters? It's a scary thought, but maybe these messages are going missing somewhere internally?

I'll see what I can do to reproduce this at my end.

**Comment by katheris** [comment]
@gracegrimwood Do the integration tests run a producer and consumer both connected to the proxy at the same time? Because trying to reproduce it today I actually only see it when I have a consumer connected while producing and I have the response filter configured. If the response filter is not configured, or I produce all my messages, then consume them all after, then I don't see the problem. And it looks it's dropping random records, not necessarily ones related to how the filter is configured.

So steps to reproduce are:

1. Start the proxy with the response filter configured
2. Create a topic
3. Start a console consumer
4. Start a console producer
5. Send records from the producer
6. The consumer doesn't see all the records

**Comment by gracegrimwood** [comment]
> Do the integration tests run a producer and consumer both connected to the proxy at the same time?

Some tests do, yes, including the sample filter integration tests. It's worth mentioning that the sample tests don't test both filters configured together, just one filter at a time. It would be simple enough to add another test for this scenario though, but I'm not sure it would have caught this particular problem as it does not send multiple messages. The integration tests for the sample just check against the one message per test to ensure the filter behaves correctly. Adding a test to check against a sustained series of messages would be entirely possible but I'm concerned that it would increase the run time of the tests significantly, so that's something we would have to weigh up.

What I've replicated matches what you've seen, that the issue occurs when a consumer is connected to the proxy at the same time as a producer (though I haven't yet got around to testing with either of the filters removed). With both sample filters configured, I connected one producer each to Kroxylicious and directly to Kafka and then did the same with consumers, and saw the issue only with the consumer connected to Kroxylicious. The consumer connected directly to Kafka managed to receive every message the producers sent, but the consumer connected to Kroxylicious stopped receiving messages after a while.

It seems odd to me that the `SampleFetchResponseFilter` works *sometimes* (or as I experienced, it worked for a while before failing) because it surely should fail always if the problem is in the filter, but I can't see a reason it would fail with specifically that filter otherwise. I might have a go at putting a no-op filter in place of the `SampleFetchResponseFilter` to see if it's what the filter is doing or if it's the presence of a filter on the response path in general that's doing it.

I'll try reproducing with one filter at a time to see if I can isolate it as you did, and I'll have a look at `SampleFetchResponseFilter` to see if there's anything peculiar there.
